File: LinkERP/ClientApp/src/app/shared/services/note.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AppConfigService } from '../../app-config.service';

@Injectable({
  providedIn: 'root'
})
export class NoteService {
  private url: string = "";
  private baseurl: string = "";

  constructor(public http: HttpClient, private appConfigService: AppConfigService) {
    this.baseurl = appConfigService.getServiceBaseUrl();
  }
  addnotes(data) {
    this.url = this.baseurl + 'api/SHARED/Note/AddNote';
    return this.http.post(this.url, data);
  }
  updatenotes(data) {
    this.url = this.baseurl + 'api/SHARED/Note/UpdateNote';
    return this.http.post(this.url, data);
  }
  getAllnotes() {

    this.url = this.baseurl + 'api/SHARED/Note/GetNotes';
    return this.http.get(this.url);
  }
  getNoteByID(ID) {
    this.url = this.baseurl + 'api/SHARED/Note/GetNoteByID/' + ID;
    return this.http.get(this.url);
  }
  deletenotessBYID(ID, LoginID) {
    this.url = this.baseurl + 'api/SHARED/Note/DeleteNoteByID/' + ID + '/' + LoginID;
    return this.http.post(this.url, null);
  }
  getAllnotetypes() {

    this.url = this.baseurl + 'api/SHARED/Note/GetNotetype';
    return this.http.get(this.url);
  }

  getNoteByRecID(ID) {
    this.url = this.baseurl + 'api/SHARED/Note/GetNoteByRecID/' + ID;
    return this.http.get(this.url);
  }
  getNoteDetailByID(ID) {
    this.url = this.baseurl + 'api/SHARED/Note/GetNoteDetailByID/' + ID;
    return this.http.get(this.url);
  }  

  deleteDeleteRecordsBYID(ID, TableName, DeletedBy)
  {
     this.url = this.baseurl + 'api/SHARED/DeleteRecords/DeleteRecordsBYID/' + ID + '/' + TableName + '/' + DeletedBy;
     return this.http.post(this.url, null);
  }
}

File: LinkERP.BLL/SHARED/NoteService.cs
using LinkERP.BLL.SHARED.Interfaces;
using LinkERP.BLL.SYS.Interfaces;
using LinkERP.DAL.SHARED.Interfaces;
using LinkERP.DAL.SYS.Interfaces;
using LinkERP.Entity.SHARED;
using LinkERP.Entity.SYS;
using System;
using System.Collections.Generic;
using System.Text;

namespace LinkERP.BLL.SHARED
{
  public   class NoteService: INoteService
    {
        INoteRepository note;
        public NoteService(INoteRepository _note)
        {
            note = _note;
        }

        public string AddNote(LBS_SYS_Notes lBS_SYS_Notes)
        {
            return note.AddNote(lBS_SYS_Notes);
        }

        public bool DeleteNoteByID(Guid ID, string DeletedBy)
        {
            return note.DeleteNoteByID(ID, DeletedBy);
        }

        public IList<LBS_SYS_Notes> GetNotes()
        {
            return note.GetNotes();
        }

        public LBS_SYS_Notes GetNoteByID(Guid ID)
        {
            return note.GetNoteByID(ID);
        }

        public string UpdateNote(LBS_SYS_Notes lBS_SYS_Notes)
        {
            return note.UpdateNote(lBS_SYS_Notes);
        }


        public IList<LBS_SYS_NoteTypes> GetNotetype()
        {
            return note.GetNotetype();
        }

        //public LBS_SYS_NoteTypes GetNotetypeByID(Guid ID)
        //{
        //    return note.GetNotetypeByID(ID);
        //}

        public IList<LBS_SYS_Notes> GetNoteByRecID(string RecID)
        {
            return note.GetNoteByRecID(RecID);
        }

        public IList<LBS_SYS_NotesDetail> GetNoteDetailByID(Guid ID)
        {
            return note.GetNoteDetailByID(ID);
        }
    }
}
